File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        x = 0 # to maintain
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[x],nums[i] = nums[i],nums[x]
                x+=1
        return nums    

        """
        Do not return anything, modify nums in-place instead.
        """
        

File: 283-move-zeroes/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
    let counter = 0;
    for (let i in nums) {
        if (nums[i] === 0) continue;
        let temp = nums[counter]
        nums[counter] = nums[i];
        nums[i] = temp;
        ++counter;
    }
};